rest_api/tests/api_test/fixtures.py

```diff
@@ -44,14 +46,214 @@
+            json_data = json.loads(error.fp.read().decode('utf-8'))
+            LOGGER.info(json_data['error']['title'])
+            LOGGER.info(json_data['error']['message']) 
+            LOGGER.info(json_data['error']['code']) 
+      
+        self.data['response'] = response['data'][0]['status'] 
+        self.data['initial_batch_length'] = initial_batch_length
+        self.data['initial_trn_length'] = initial_transaction_length
+        self.data['expected_batch_length'] = initial_batch_length + length_batches
+        self.data['expected_trn_length'] = initial_transaction_length + length_transactions
+        return self.data   
+    
+    def create_intkey_transaction(self, verb, deps, count, signer):
+            words = random_word_list(count)
+            name=random.choice(words)    
+            payload = IntKeyPayload(
+                verb=verb,name=name,value=1)
+
+            addr = make_intkey_address(name)
+            data = self.get_txns_data(addr,deps, payload)
+            return data
+        
+    def create_invalid_intkey_transaction(self, verb, deps, count, signer, invalidtye):
+        words = random_word_list(count)
+        name=random.choice(words) 
+        
+        if invalidtye=="addr":  
+            payload = IntKeyPayload(
+                verb=verb,name=name,value=1)
+            
+            INVALID_INTKEY_ADDRESS_PREFIX = hashlib.sha512(
+            'invalid'.encode('utf-8')).hexdigest()[0:6]
+        
+            addr = INVALID_INTKEY_ADDRESS_PREFIX + hashlib.sha512(
+                name.encode('utf-8')).hexdigest()[-64:]
+            
+        elif invalidtye=="min":   
+            payload = IntKeyPayload(
+                verb=verb,name=name,value=-1)
+            addr = make_intkey_address(name)
+        
+        elif invalidtye=="str":    
+            payload = IntKeyPayload(
+                verb=verb,name=name,value="str")
+            addr = make_intkey_address(name)
+            
+        elif invalidtye=="max":    
+            payload = IntKeyPayload(
+                verb=verb,name=name,value=4294967296)
+            addr = make_intkey_address(name)
+            
+        elif invalidtye=="attr":    
+            payload = IntKeyPayload(
+                verb="verb",name=name,value=1)
+            addr = make_intkey_address(name)
+
+        data = self.get_txns_data(addr,deps, payload)
+        return data
+   
+    def get_txns_data(self, addr, deps, payload):
+    
+        header = TransactionHeader(
+            signer_public_key=self.signer.get_public_key().as_hex(),
+            family_name='intkey',
+            family_version='1.0',
+            inputs=[addr],
+            outputs=[addr],
+            dependencies=deps,
+            payload_sha512=payload.sha512(),
+            batcher_public_key=self.signer.get_public_key().as_hex())
+    
+        header_bytes = header.SerializeToString()
+    
+        signature = self.signer.sign(header_bytes)
+    

```

Comment:
Transaction creation is done in payload file ...then why use Transaction header here....create there and call here....Please rework